tallycash/extension
# 🥗 Resolve CPU issues from `blocks` bloat in Redux store

When having Tally Ho installed for a significant period of time, the CPU usage
increases to pretty extreme levels. On the Chrome Store/production install I
have on Brave, I see 70,922 blocks and 114,214 balance entries saved in
IndexedDB. These blocks and balance entries accumulate over time. While
IndexedDB isn't the culprit of the CPU issues, I believe a significant portion
is making its way into the Redux store. The massive Redux store I believe is
the culprit for extreme CPU usage.

This PR removes the dependance on storing blocks in Redux. Their only
current use is for deriving an activity's timestamp. That approach is replaced
by [adding the transaction's associated block's timestamp to the transaction
annotation via the `EnrichmentService`](https://github.com/tallycash/extension/pull/1363/commits/161ec56f5a4f81572be3d401cd18f0eac410c946). Since individual blocks are no longer
used in the FE, this PR removes saving them in the networks slice at all (aside
from the latest `blockHeight`). By no longer including a huge number of blocks in
the Redux store, this change should significantly ameliorate the CPU issue.

#1351

### Testing instructions
1. Test to see that timestamps work on activities
  - On the home tab
  - On the activity detail panel
2. With Redux dev tools, ensure that no `blocks` are being added to the Redux
store (networks → evm["1"] → blocks)
3. Test the migration by checking out an older version of Tally Ho, let the block
data and activities load in, and then update to this branch 

Closes #1351
